research/delf/delf/python/training/README.md

```diff
@@ -1,29 +1,134 @@
+that the GLDv2 images have been downloaded to the `gldv2_dataset` folder, the script can be run as follows:
+```
+python3 build_image_dataset.py \
+    --train_csv_path=gldv2_dataset/train/train.csv \
+    --train_clean_csv_path=gldv2_dataset/train/train_clean.csv \
+    --train_directory=gldv2_dataset/train/*/*/*/ \
+    --test_csv_path=gldv2_dataset/train/test.csv \
+    --test_directory=gldv2_dataset/test/*/*/*/ \
+    --output_directory=gldv2_dataset/tfrecord/ \
+    --num_shards=128 \
+    --generate_train_validation_splits \
+    --validation_split_size=0.2
+```
+The [`build_image_dataset.py`](./build_image_dataset.py) takes the following parameters:
+* `train_csv_path` - Path of the CSV file referencing images from the TRAIN dataset.
+* `train_clean_csv_path` - (Optional) Path of the CSV file referencing images from the *clean*
+TRAIN dataset. If provided:
+  * Images will be filered by only keeping the ones listed in this file
+  * Images will also be relabeled in order to guarantee a continuous sequence of labels. The
+  script will output a `[OUTPUT_DIRECTORY]/relabeling.csv` file containing the mapping between the
+  new labels and the old labels.
+* `train_directory` - Training data directory.
+* `test_csv_path` - (Optional) Path of the CSV file referencing images from the TEST dataset. If
+None or absent, TFRecords for the images in the TEST dataset are not generated.
+* `test_directory` - (Optional) Testing data directory. Required only if `test_csv_path` is not
+None.
+* `output_directory ` - Directory where the TFRecord files and relabeling file will be written.
+* `num_shards` - Number of shards in which each split (TRAIN, VALIDATION, TEST) will be broken.
+* `generate_train_validation_splits` - (Optional) Whether to split the TRAIN dataset into TRAIN
+and VALIDATION splits.
+* `validation_split_size` - (Optional) The size of the VALIDATION split as a fraction of the 
+TRAIN dataset.
+* `seed` - (Optional) The seed to be used while shuffling the TRAIN dataset when generating the
+TRAIN and VALIDATION splits. Recommended for splits reproducibility purposes.
+
+The TFRecord files written in the `OUTPUT_DIRECTORY` will be prefixed as follows:
+* TRAIN split: `train-*`
+* VALIDATION split: `validation-*`
+* TEST split: `test-*`
+
+*Please note that due to the large size of the GLDv2 dataset, the generation of the TFRecord 
+files can take up to 12 hours and up to 500 GB of space disk.*
+
+## Running the Training
+
+Assuming the TFRecord files were generated in the `gldv2_dataset/tfrecord/` directory, running 
+the following command should start training a model:
 
 ```sh
 python3 tensorflow_models/research/delf/delf/python/training/train.py \
-  --train_file_pattern=/tmp/gld_tfrecord/train* \
-  --validation_file_pattern=/tmp/gld_tfrecord/train* \
+  --train_file_pattern=gldv2_dataset/tfrecord/train* \
+  --validation_file_pattern=gldv2_dataset/tfrecord/validation* \
   --debug
```

Comment:
maybe remove the `debug`? Or else mention that this command is run in debug mode, and that one could remove it to run the large-scale real training.

Reply:
I removed `debug`, it didn't make sense to keep it there, you're right.